vpc/service/auth.go

```diff
@@ -57,12 +57,34 @@ func (vpcService *titusVPCAgentServiceAuthFuncOverride) AuthFuncOverride(ctx con
 	if err != nil {
 		return ctx, errors.Wrap(err, "Cannot parse remote certificate")
 	}
-	_, err = cert.Verify(x509.VerifyOptions{
+	err = vpcService.validateCert(cert)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to verify client cert: %w", err)
+	}
+	return ctx, nil
+}
+
+func (vpcService *titusVPCAgentServiceAuthFuncOverride) validateCert(cert *x509.Certificate) error {
+	// This first check only validates the chain of trust
+	_, err := cert.Verify(x509.VerifyOptions{
 		Roots:     vpcService.TitusAgentCACertPool,
 		KeyUsages: []x509.ExtKeyUsage{x509.ExtKeyUsageClientAuth},
 	})
 	if err != nil {
-		return nil, errors.Wrap(err, "Unable to verify client cert")
+		return err
 	}
-	return ctx, nil
+	// The cert.Verify function is relativly limited in what it can verify.
+	// It only can take in a single string and match it against DNS/CN.
+	// We know we will need to match against a variety of incoming names,
+	// so we check everything in a loop and return an error if nothing matched.
+	if vpcService.ValidCNRegex.MatchString(cert.Subject.CommonName) {
+		return nil
+	}
+	for _, san := range cert.DNSNames {
+		if vpcService.ValidCNRegex.MatchString(san) {
+			return nil
+		}
+	}
+
+	return fmt.Errorf("Client certificate's CN: %q and SANS: %q failed to match our allow list: %v", cert.Subject.CommonName, cert.DNSNames, &vpcService.ValidCNRegex)
```

Comment:
Dumb question, why is this a pointer?

Reply:
Also, an aside, does this error get logged?

Reply:
regex.compile returns a pointer to a struct.
When dereferenced like this it prints the actual regex instead of a hard to read struct representation.
I'll change it to just be `String()` to be more obvious.

It appears both client and server side. From vpctool:
```
FATA[0000] Failed                                        command=setup-instance error="rpc error: code = Unknown desc = Unable to verify client cert: Client certificate's CN: \"\" and SANS: [\"main01cell001.titusagent.dyntest.netflix.net\" \"main01cell001.titusagent.us-west-2.dyntest.netflix.net\" \"titusagent.dyntest.netflix.net\" \"titusagent.test.netflix.net\" \"titusagent.us-west-2.dyntest.netflix.net\" \"titusagent.us-west-2.test.netflix.net\" \"main01cell001.titusagent.us-west-2.test.netflix.net\" \"main01cell001.titusagent.test.netflix.net\" \"titusagent.app.us-west-2.test.cloud.netflix.net\"] failed to match our allow list: titusagentWRONG\\..*"
```

And from the vpcservice:
```
ERRO[0002] finished unary call with code Unknown         error="Unable to verify client cert: Client certificate's CN: \"\" and SANS: [\"main01cell001.titusagent.dyntest.netflix.net\" \"main01cell001.titusagent.us-west-2.dyntest.netflix.net\" \"titusagent.dyntest.netflix.net\" \"titusagent.test.netflix.net\" \"titusagent.us-west-2.dyntest.netflix.net\" \"titusagent.us-west-2.test.netflix.net\" \"main01cell001.titusagent.us-west-2.test.netflix.net\" \"main01cell001.titusagent.test.netflix.net\" \"titusagent.app.us-west-2.test.cloud.netflix.net\"] failed to match our allow list: titusagentWRONG\\..*" grpc.code=Unknown grpc.method=ProvisionInstanceV3 grpc.request.deadline="2020-10-07T18:25:47Z" grpc.service=com.netflix.titus.executor.vpc.TitusAgentVPCService grpc.start_time="2020-10-07T18:23:47Z" grpc.time_ms=0.302 origin=grpc peer.address="100.78.155.109:34696" span.kind=server system=grpc
```

Should be enough data to figure out who was connecting and why it failed.